src/components/course/tests/EnrollButton.test.jsx

```diff
@@ -196,10 +233,67 @@ describe('<EnrollButton />', () => {
           initialAppState={initialAppState}
           initialCourseState={courseState}
           initialUserSubsidyState={initialUserSubsidyState}
+          initialReduxStore={initialReduxStore}
         />,
       );
 
       expect(screen.queryByText('View Course'));
     });
   });
+  describe('getEnrollmentUrl', () => {
+    const noSubscriptionEnrollmentInputs = {
+      enterpriseConfig: {
+        uuid: 'foo',
+      },
+      key: 'bar',
+      offers: [{ code: 'bearsRus', catalog: 'bears' }],
+      offersCount: 2,
+      offersLoading: false,
+      sku: 'xkcd',
+      catalogList: ['bears'],
+      location: { search: 'foo' },
+    };
+    const enrollmentInputs = {
+      ...noSubscriptionEnrollmentInputs,
+      subscriptionLicense: {
+        uuid: 'yes',
+      },
+    };
+    it('Subscription: returns an lms url with correct querystring', () => {
+      const url = getEnrollmentUrl(enrollmentInputs);
+      expect(url).toContain(process.env.LMS_BASE_URL);
+      expect(url).toContain(enrollmentInputs.enterpriseConfig.uuid);
+      expect(url).toContain(enrollmentInputs.key);
+      expect(url).toContain(enrollmentInputs.subscriptionLicense.uuid);
+    });
+    it('No subscription with offers for course: returns an ecommerce url with correct querystring', () => {
+      const url = getEnrollmentUrl(noSubscriptionEnrollmentInputs);
+      expect(url).toContain(process.env.ECOMMERCE_BASE_URL);
+      expect(url).toContain(noSubscriptionEnrollmentInputs.sku);
+      expect(url).toContain(noSubscriptionEnrollmentInputs.offers[0].code);
+      expect(url).toContain(enrollmentInputs.key);
+    });
+    it('No subscription with no offers for course: returns ecommerce url with correct querystring', () => {
+      const url = getEnrollmentUrl({ ...noSubscriptionEnrollmentInputs, catalogList: ['foo'] });
+      expect(url).toContain(process.env.ECOMMERCE_BASE_URL);
+      expect(url).toContain(noSubscriptionEnrollmentInputs.sku);
+      expect(url).toContain(enrollmentInputs.key);
+      expect(url).not.toContain('code');
+    });
+    it('No subscription no offers: returns an ecommerce url with correct querystring', () => {
+      const url = getEnrollmentUrl({ ...noSubscriptionEnrollmentInputs, offersCount: 0 });
+      expect(url).toContain(process.env.ECOMMERCE_BASE_URL);
```

Comment:
I wonder in general for these tests, how precise do we need to be with the urls? In other words, should we be testing for a precise match? if possible, shall we do that? such as `expect(url).toEqual/toBe(expectedUrl)` ?

Reply:
So, the base url is coming from an env variable, and could thus change in different environments/test runners. But yeah, the rest of it could be a more precise match. I think it's a trade off between a perhaps more accurate test, but also a much more brittle test.